### PR TITLE
Removed unneccessary renames for gocql

### PR DESCRIPTION
**Pull request by MadVikingGod**
The Release of metrics v0.32.1 means that the prometheus names are automatically renamed.  This change removes unnecessary renames.

Blocked by https://github.com/open-telemetry/opentelemetry-go/pull/3228, and release.

### CODE REVIEW

File: instrumentation/github.com/gocql/gocql/otelgocql/example/client.go [hunk 1]
```diff
@@ -203,7 +195,7 @@ func initMetrics() error {
 	http.Handle("/", promhttp.Handler())
 	log.Print("Serving metrics at :2222/")
```
Comment (MrAlias) [inline instrumentation/github.com/gocql/gocql/otelgocql/example/client.go:196]:
```suggestion
	log.Print("Serving metrics at :2223/")
```

File: instrumentation/github.com/gocql/gocql/otelgocql/example/client.go [hunk 2]
```diff
@@ -129,12 +129,8 @@ func main() {
 
 func views() ([]view.View, error) {
 	var vs []view.View
-	// TODO: Remove renames when the Prometheus exporter natively supports
-	// metric instrument name sanitation
-	// (https://github.com/open-telemetry/opentelemetry-go/issues/3183).
 	v, err := view.New(
 		view.MatchInstrumentName("db.cassandra.queries"),
-		view.WithRename("db_cassandra_queries"),
 	)
```
Comment (MrAlias) [inline instrumentation/github.com/gocql/gocql/otelgocql/example/client.go:134]: These "blank action" views could be unified into a single view that matches `"*"`

File: instrumentation/github.com/gocql/gocql/otelgocql/example/client.go (hunk 1 above)
Comment (MrAlias) [reply instrumentation/github.com/gocql/gocql/otelgocql/example/client.go:196]: The README.md also needs updating if this is changing.

**Comment by MadVikingGod** [comment]
This will fix the first set of changes in #2796

**Comment by MadVikingGod** [comment]
Closing in favor of #2960.

**Pull request closed without merging**